ntp-proto/src/packet/mod.rs

```diff
@@ -1074,30 +1159,42 @@ impl<'a> NtpPacket<'a> {
         }
     }
 
+    pub fn kiss_code(&self) -> ReferenceId {
+        match self.header {
+            NtpHeader::V3(header) => header.reference_id,
+            NtpHeader::V4(header) => header.reference_id,
+            #[cfg(feature = "ntpv5")]
+            // Kiss code in ntpv5 is the first four bytes of the server vookie
+            NtpHeader::V5(header) => {
+                ReferenceId::from_bytes(header.server_cookie.0[..4].try_into().unwrap())
+            }
+        }
+    }
+
     pub fn is_kiss(&self) -> bool {
```

Comment:
If `Self::kiss_code` returns an option this could be removed or made private

Reply:
see above